### ISSUE TITLE
Remote Schemas: Can't return interface type when using Remote Schema permissions

### ISSUE THREAD

**Issue by aaka3207** [hasura/graphql-engine#8720]
### Version Information

Server Version: **2.8.3**
CLI Version (for CLI related issue): **v2.4.0**

### What is the current behaviour?
I have a remote schema with a field that resolves to an interface type, where there are some shared fields (such as id, name, and type), and then child fields that inherit from that field. It's much cleaner to do this than to implement unions since most of the child fields have only slightly different fields.

I'm able to query this in the console (when using the admin role). In the example below **schema** is a field of interface type _IWorkspacePropertySchema_

![image](https://user-images.githubusercontent.com/16450271/181145745-fff8cd0b-6e3f-48f8-99a8-f709b03953e8.png)

But when trying to add remote schema permissions for my resolver, it throws an error saying that it couldn't find the interface type:
`{ "path": "$.args[0].args", "error": "Could not find type with name \"IWorkspacePropertySchema\"", "code": "remote-schema-error" }
`

### How to reproduce the issue?

1.create a resolver with an object type containing a field that is an interface
2. query for that resolver in the graphql console
3. create remote schema permissions for that resolver using any other role.

**Comment by harishnair96** [comment]
This seems to be a console issue where it doesn't pass the schema correctly to the backend when adding permissions with GraphQL interfaces. However, the good news is that you will be able to get it working by directly applying the changes to the metadata. Take a look at the example below.

If I have a remote schema with the following GraphQL schema - 
```graphql
  interface Item {
    price: Int
    name: String
  }

  type Book implements Item {
    title: String
    author: String
    price: Int
    name: String
  }

  type Query {
    books: [Book]
    items: [Item]
  }
```

If I want to allow a role `user` to query the field `items`, I can do it by applying the following metadata -

```yaml
- name: books-schema
  definition:
    url_from_env: REMOTE_SCHEMA_ENDPOINT
    timeout_seconds: 60
  comment: ""
  permissions:
    - role: user
      definition:
        schema: |
          type Book implements Item {
          	author: String
          	name: String
          	price: Int
          	title: String
          }
          interface Item {
            price: Int
            name: String
          }
          type Query {
          	books: [Book]
            items: [Item]
          }
```

Please don't forget to add the `implements` syntax to all the types that implement the interface or else the query will not work.

**Comment by harishnair96** [comment]
Issue still exists even in `v2.13.0-beta.1`

**Comment by manasag** [comment]
This have been fixed as per this [commit](https://github.com/hasura/graphql-engine/commit/e13ff5babe84436aee32ebfa880101c34bbca7f9), released in v2.15. Closing this issue.
